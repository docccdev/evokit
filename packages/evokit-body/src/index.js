import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
