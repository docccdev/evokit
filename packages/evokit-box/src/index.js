// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
