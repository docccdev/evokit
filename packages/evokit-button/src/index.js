import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
