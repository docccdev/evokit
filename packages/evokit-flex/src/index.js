import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'display',
    'content',
    'items',
    'direction',
    'wrap',
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'order',
]);

Flex.Item = FlexItem;
