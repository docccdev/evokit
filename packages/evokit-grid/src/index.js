import { createBlock } from 'evokit';

export const Grid = createBlock('div', 'grid', [
    'align',
    'column',
    'direction',
    'display',
    'divider',
    'divider-column',
    'height',
    'valign',
    'wrap',
    ['divider-indent', [
        'divider-indent-x',
        'divider-indent-y',
    ]],
    ['indent', [
        'indent-x',
        'indent-y',
    ]],
]);

export const GridItem = createBlock('div', 'grid__item', [
    'display',
    'order',
    'width',
]);

Grid.Item = GridItem;
