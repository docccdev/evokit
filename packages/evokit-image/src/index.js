import { createBlock } from 'evokit';

export const Image = createBlock('img', 'image', [
    'align',
    'alt',
    'border',
    'border-bottom',
    'border-left',
    'border-lr',
    'border-right',
    'border-tb',
    'border-top',
    'display',
    'fit',
    'mirror',
    'valign',
    'width',
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);
