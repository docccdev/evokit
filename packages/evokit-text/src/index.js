import { createBlock } from 'evokit';

export const Text = createBlock('span', 'text', [
    'align',
    'color',
    'display',
    'lheight',
    'size',
    'style',
    'transform',
    'valign',
    'weight',
    'wrap',
]);
