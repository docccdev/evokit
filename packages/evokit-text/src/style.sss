@import './mixins.sss'

@custom-media --ek-text-media-small only screen and (min-width: 480px)
@custom-media --ek-text-media-medium only screen and (min-width: 768px)
@custom-media --ek-text-media-large only screen and (min-width: 960px)
@custom-media --ek-text-media-wide only screen and (min-width: 1200px)
@custom-media --ek-text-media-huge only screen and (min-width: 1400px)

:root
    // base
    --ek-text-line-height: 1em
    // prop 'text-size'
    --ek-text-size-small: 11px
    --ek-text-size-default: 13px
    --ek-text-size-big: 15px
    --ek-text-size-h1: 24px
    --ek-text-size-h2: 22px
    --ek-text-size-h3: 20px
    --ek-text-size-h4: 18px
    --ek-text-size-h5: 16px
    --ek-text-size-h6: 14px
    // prop 'text-lheight'
    --ek-text-line-height-default: var(--ek-text-line-height)
    --ek-text-line-height-medium: 1.25em
    --ek-text-line-height-large: 1.5em

.text
    display: inline
    line-height: var(--ek-text-line-height)

/**
 * Display:
 *
 *  - text_display_inline
 *  - text_display_none
 *
 **/

@each $value in (inline, none)
    @mixin media .text_display_$(value)
        display: $value

/**
 * Font size:
 *
 *  - text_size_*
 *
 **/

@each $value in (small, default, big, h1, h2, h3, h4, h5, h6)
    @mixin media .text_size_$(value)
        font-size: var(--ek-text-size-$(value))

/**
 * Font weight:
 *
 *  - text_weight_*
 *
 **/

@each $value in (default, thin, light, medium, bold, black)
    @mixin media .text_weight_$(value)
        @if $value == default
            font-weight: 400
        @else if $value == thin
            font-weight: 100
        @else if $value == light
            font-weight: 300
        @else if $value == medium
            font-weight: 500
        @else if $value == bold
            font-weight: 700
        @else if $value == black
            font-weight: 900

/**
 * Align:
 *
 *  - text_align_*
 *
 **/

@each $value in (left, center, right, justify)
    @mixin media .text_align_$(value)
        display: block
        text-align: $value

/**
 * Valign:
 *
 *  - text_valign_*
 *
 **/

@each $value in (baseline, bottom, middle, sub, super, text-bottom, text-top, top)
    @mixin media .text_valign_$(value)
        vertical-align: $value

/**
 * Transform:
 *
 *  - text_transform_*
 *
 **/

@each $value in (none, capitalize, lowercase, uppercase)
    @mixin media .text_transform_$(value)
        text-transform: $value

/**
 * Style:
 *
 *  - text_style_*
 *
 **/

@each $value in (italic, strike, underline, overline)
    @mixin media .text_style_$(value)
        @if $value == italic
            font-style: italic
        @else if $value == strike
            text-decoration: line-through
        @else if $value == underline
            text-decoration: underline
        @else if $value == overline
            text-decoration: overline

/**
 * Line height:
 *
 *  - text_lheight_*
 *
 **/

@each $value in (default, medium, large)
    @mixin media .text_lheight_$(value)
        line-height: var(--ek-text-line-height-$(value))

/**
 * Wrap:
 *
 *  - text_wrap_*
 *
 **/

@each $value in (nowrap, break, ellipsis, three-line, two-line)
    @mixin media .text_wrap_$(value)
        @if $value == nowrap
            white-space: nowrap
        @else if $value == break
            word-wrap: break-word
            word-break: break-word
            hyphens: auto
        @else if $value == ellipsis
            display: block
            overflow: hidden
            text-overflow: ellipsis
            white-space: nowrap
        @else if $value == two-line
            display: block
            overflow: hidden
            max-height: calc(var(--ek-text-line-height) * 2)

            &[class*='text_lheight_medium']
                max-height: calc(var(--ek-text-line-height-medium) * 2)

            &[class*='text_lheight_large']
                max-height: calc(var(--ek-text-line-height-large) * 2)

        @else if $value == three-line
            display: block
            overflow: hidden
            max-height: calc(var(--ek-text-line-height) * 3)

            &[class*='text_lheight_medium']
                max-height: calc(var(--ek-text-line-height-medium) * 3)

            &[class*='text_lheight_large']
                max-height: calc(var(--ek-text-line-height-large) * 3)
