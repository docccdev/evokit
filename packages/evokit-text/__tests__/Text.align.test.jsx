import React from 'react';
import { shallow } from 'enzyme';
import { Text } from '../src';

describe('<Text text-align="..." />', () => {
    it('props text-align', () => {
        const wrapper = shallow(<Text text-align='value' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<span class="ek-text ek-text_align_value"></span>');
    });
});
