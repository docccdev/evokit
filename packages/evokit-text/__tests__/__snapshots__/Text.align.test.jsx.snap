// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text text-align="..." /> props text-align 1`] = `
<span
  className="ek-text ek-text_align_value"
/>
`;
